### PR TITLE
Rel 507460 update disclaimer helpers to rest

### PR DESCRIPTION
**Pull request by aarongilbert123**

### CODE REVIEW

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 1]
```diff
@@ -309,20 +389,38 @@ private int GetDisclaimerSolutionConfigurationObjectTypeId(int workspaceId)
 			return objectTypeId;
 		}
 
-		private int GetDisclaimerLayoutId()
+		private async Task<int> GetDisclaimerLayoutId(int workspaceId)
```
Comment (TommyBigNose) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:392]: - [ ] End function name in Async

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 2]
```diff
@@ -252,26 +306,37 @@ private int GetDisclaimerObjectTypeId(int workspaceId)
 			return objectTypeId;
 		}
 
-		private int GetWorkspaceId(string workspaceName)
+		private async Task<int> GetWorkspaceId(string workspaceName)
```
Comment (TommyBigNose) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:309]: - [ ] End function name in Async

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 3]
```diff
@@ -225,24 +264,39 @@ private async Task CreateDisclaimerRdoAsync(int objectTypeId, int layoutId, int
 			}
 		}
 
-		private int GetDisclaimerObjectTypeId(int workspaceId)
+		private async Task<int> GetDisclaimerObjectTypeId(int workspaceId)
```
Comment (TommyBigNose) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:267]: - [ ] End function name in Async

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 4]
```diff
@@ -332,20 +430,38 @@ private int GetDisclaimerLayoutId()
 			return layoutId;
 		}
 
-		private int GetDisclaimerSolutionConfigurationLayoutId()
+		private async Task<int> GetDisclaimerSolutionConfigurationLayoutId(int workspaceId)
```
Comment (TommyBigNose) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:433]: - [ ] End function name in Async

**Review by TommyBigNose — APPROVED** [review]

**Review by TommyBigNose — CHANGES_REQUESTED** [review]
Oh, sorry, can you update the CHANGELOG with the past few tickets associated with the REST updates?

**Review by TommyBigNose — DISMISSED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 5]
```diff
@@ -252,26 +306,37 @@ private int GetDisclaimerObjectTypeId(int workspaceId)
...
+					start = 1,
+					length = 25
+				};
+
+				string queryPayload = JsonConvert.SerializeObject(queryPayloadObject);
+				HttpResponseMessage queryResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ObjectManager.QuerySlimUrl, queryPayload);
+				if (!queryResponse.IsSuccessStatusCode)
 				{
-					throw new Exception($"Unable to find workspace with the name: {workspaceName}");
+					throw new Exception("Failed to Query for Disclaimer Solution Configuration Object Type");
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:334]: - [ ] Fix error message

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs [hunk 6]
```diff
@@ -252,26 +306,37 @@ private int GetDisclaimerObjectTypeId(int workspaceId)
 			return objectTypeId;
 		}
 
-		private int GetWorkspaceId(string workspaceName)
+		private async Task<int> GetWorkspaceIdAsync(string workspaceName)
```
Comment (calimetikcura) [inline CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs:309]: Should we move this API call to the Workspace helper?

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs (hunk 6 above)
Comment (aarongilbert123) [reply CSharp/DevVmPowershell/Helpers/Implementations/DisclaimerAcceptanceHelper.cs]: Done

**Review by calimetikcura — APPROVED** [review]

**Review by TommyBigNose — APPROVED** [review]

**Pull request merged**